calico/etcddriver/driver.py

```diff
@@ -678,6 +711,9 @@ def watch_etcd(self, next_index, event_queue, stop_event):
                                 # deleted, that implies the ready flag is gone
                                 # too; resync rather than generating deletes
                                 # for every key.
+                                #PLW: Shouldn't we treat just as if the ready
+                                # flag got nuked in a response? In other
+                                # words, put a nuke of /ready onto the queue?
```

Comment:
I think it's better to break here and let the resync thread spot that the watcher is gone.  If we carry on polling, the watcher won't die until some time later (but we know that all its events will be being ignored by the resync thread).